### PR TITLE
Add MAIN_PROJECT check for test and install options

### LINKED ISSUE #2513: Including CTest in the top-level CMakeLists.txt sets BUILD_TESTING=ON for parent projects
#### What is the issue you have?

BUILD_TESTING is unset in the main project.
nlohmann_json is included in a main project using FetchContent.
JSON_BuildTests=OFF is set in the main project.
nlohmann_json's top-level CMakeLists.txt includes CTest regardless.
BUILD_TESTING=ON is now set in the main project.

#### Please describe the steps to reproduce the issue.

```
message(STATUS "before BUILD_TESTING := ${BUILD_TESTING}") # prints ""
include(FetchContent)
FetchContent_Declare(
  nlohmann_json
  GIT_REPOSITORY https://github.com/nlohmann/json.git
  GIT_TAG develop)
set(JSON_BuildTests OFF)
FetchContent_MakeAvailable(nlohmann_json)
message(STATUS "after BUILD_TESTING := ${BUILD_TESTING}") # prints ON
```

#### What is the expected behavior?

CTest should only be included if nlohmann_json is the main project.

#### And what is the actual behavior instead?

CTest is included although nlohmann_json is added as subproject.

#### Which version of the library did you use?

- [ ] latest release version 3.9.1
- [ ] other release - please state the version: ___
- [x] the `develop` branch

### PR DESCRIPTION
**Pull request by globberwops**
This resolves #2513 by checking if this is the main project in a CMake project tree.
`JSON_BuildTests` and `JSON_Install` will only be set to `ON` if this is the main project.
CTest will only be included, if `JSON_BuildTests` is set to `ON`.
`BUILD_TESTING` will therefore only be set, if this is the main project.

## Pull request checklist

Read the [Contribution Guidelines](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md) for detailed information.

- [x]  Changes are described in the pull request, or an [existing issue is referenced](https://github.com/nlohmann/json/issues).
- [x]  The test suite [compiles and runs](https://github.com/nlohmann/json/blob/develop/README.md#execute-unit-tests) without error.
- [x]  [Code coverage](https://coveralls.io/github/nlohmann/json) is 100%. Test cases can be added by editing the [test suite](https://github.com/nlohmann/json/tree/develop/test/src).
- [x]  The source code is amalgamated; that is, after making changes to the sources in the `include/nlohmann` directory, run `make amalgamate` to create the single-header file `single_include/nlohmann/json.hpp`. The whole process is described [here](https://github.com/nlohmann/json/blob/develop/.github/CONTRIBUTING.md#files-to-change).

### CODE REVIEW

**Comment by ayushsinghsrinet** [comment]
Can you teach me c++

**Comment by YarikTH** [comment]
For reference, here is the same purpose code from [{fmt}](https://github.com/fmtlib/fmt/blob/master/CMakeLists.txt)
```cmake
# Determine if fmt is built as a subproject (using add_subdirectory)
# or if it is the master project.
set(MASTER_PROJECT OFF)
if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  set(MASTER_PROJECT ON)
  message(STATUS "CMake version: ${CMAKE_VERSION}")
endif ()
...
...
...
# Options that control generation of various targets.
option(FMT_DOC "Generate the doc target." ${MASTER_PROJECT})
option(FMT_INSTALL "Generate the install target." ${MASTER_PROJECT})
option(FMT_TEST "Generate the test target." ${MASTER_PROJECT})
option(FMT_FUZZ "Generate the fuzz target." OFF)
option(FMT_CUDA_TEST "Generate the cuda-test target." OFF)
option(FMT_OS "Include core requiring OS (Windows/Posix) " ON)
```

So it might be a good idea at least copy explanatory comment and maybe use `CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR` check instead of `CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,6 +6,14 @@ cmake_minimum_required(VERSION 3.1)
 ##
 project(nlohmann_json VERSION 3.9.1 LANGUAGES CXX)
 
+##
+## MAIN_PROJECT CHECK
```
Comment (nlohmann) [inline CMakeLists.txt:10]: Please add documentation similar to https://github.com/nlohmann/json/pull/2514#issuecomment-743769382.

**Review by nlohmann — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -101,9 +110,8 @@ CONFIGURE_FILE(
 ## TESTS
 ## create and configure the unit test target
 ##
-include(CTest) #adds option BUILD_TESTING (default ON)
-
-if(BUILD_TESTING AND JSON_BuildTests)
+if (JSON_BuildTests)
+    include(CTest) #adds option BUILD_TESTING (default ON)
```
Comment (nlohmann) [inline CMakeLists.txt:114]:
```suggestion
    include(CTest)
```

This comment is not needed any more.

**Review by nlohmann — CHANGES_REQUESTED** [review]
A last change request, otherwise looks good to me.

**Review by nlohmann — APPROVED** [review]
Looks good to me.

**Comment by nlohmann** [comment]
Thanks!

**Comment by globberwops** [comment]
@nlohmann Thank you 👍🏻

**Pull request merged**